admin/section/class-convertkit-settings-general.php

```diff
@@ -215,14 +223,32 @@ public function debug_callback() {
 		}
 
 		echo sprintf( // WPCS: XSS OK
-			'<input type="checkbox" class="" id="debug" name="%s[debug]"  %s />%s',
+			'<label><input type="checkbox" class="" id="debug" name="%s[debug]"  %s />%s</label>',
 			$this->settings_key,
 			$debug,
 			__( 'Save connection data to a log file.','convertkit' )
 		);
 
 	}
 
+	/**
+	 * Renders the input for no_scripts setting
+	 */
+	public function no_scripts_callback() {
+
+		$no_scripts = '';
+		if ( isset( $this->options['no_scripts'] ) && 'on' === $this->options['no_scripts'] ) {
+			$no_scripts = 'checked';
+		}
```

Comment:
Assigning `$no_scripts = ''` and checking for isset() in the condition below seems repetitive, is that the PHP way?

Reply:
In this case, it's because we either want `$no_scripts` to be an empty string (`''`), or the string `checked`. That's because it's going in the html to check the box (or not) based on the current value from the database.

The `isset()` isn't checking `$no_scripts`; it's checking the `no_scripts` key in the `$this->options` array.

Make sense?

Reply:
Yeah, I understood that. It just felt weird looking at it. In ruby we'd do something like this:

```
$no_script = $this->options.fetch("no_scripts", "") == "on"
```
^ pseudo code above

And then we would use the boolean value.

Reply:
Yeah that makes sense. Working on some improvements to this 👏